devito/ir/support/utils.py

```diff
@@ -80,6 +80,7 @@ def build_iterators(mapper):
     for k, v in mapper.items():
         for d, offs in v.items():
             if d.is_Stepping:
+                offs = sorted(offs, key=lambda x: -float("inf") if x == 0 else x)
```

Comment:
update me on what's going on here ?

Reply:
smells of hack

Reply:
You can check the thread on slack 🤣  but basically just makes sure offsets are ordered the way sympy orders additions based on indices. So it's not a hack but just a compatibility reordering.